src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -557,7 +557,15 @@ private static ExpressionSyntax SkipParensAndNullSuppressions(ExpressionSyntax e
             {
                 if (constantValueOpt == null)
                 {
-                    diagnostics.Add(ErrorCode.ERR_ConstantExpected, patternExpression.Location);
```

Comment:
I think the suggestion from https://github.com/dotnet/roslyn/pull/64741#discussion_r996202027 was to not introduce any new logic, and change this to just use the new error code you added, passing in the `inputType`.

Reply:
I've removed the check for user defined conversions. However a few tests where the input type was unknown you'd end up with messages like `Constant value of type '?' expected`. Which I dont think improves upon the existing behavior.

For this reason I've now implemented to conditionally return the original `ERR_ConstantExpected` when the inputType is an error symbol. If this is undesirable please let me know.